static/js/translations.js

```diff
@@ -1,97 +1,25 @@
 // commonjs code goes here
-
 import i18next from 'i18next';
-import XHR from 'i18next-xhr-backend';
-import LngDetector from 'i18next-browser-languagedetector';
-import Cache from 'i18next-localstorage-cache';
-import localstorage from './localstorage';
 
 window.i18n = i18next;
-
-// Add those keys in this list which are received from the backend
-// and are translated by calling i18n.t function on variables. For example,
-// i18n.t(receivedFromBackend);
-var toBeTranslated = [  // eslint-disable-line no-unused-vars
-    // The Emoji type name for the "text" emojiset choice
-    i18n.t('Plain text'),
-];
-
-function loadPath(languages) {
-    var language = languages[0];
-    if (language.indexOf('-') >= 0) {
-        language = language.replace('-', '_');  // Change zh-Hans to zh_Hans.
-    }
-
-    return '/static/locale/' + language + '/translations.json';
+var language = page_params.default_language;
+if (language.indexOf('-') >= 0) {
+    language = language.replace('-', '_');  // Change zh-Hans to zh_Hans.
```

Comment:
The reasoning for this language string conversion is: At first, I was thinking language provided in `init()` will act as a key and it will use corresponding `resource` from`addResources`, but it wasn't working for `zh-Hans` even though we handle this in the backend. Hence added this, though I'm not sure why it wasn't working for `zh-Hans` but worked for `zh_Hans`.

Reply:
@umairwaheed can you explain what's going on here?

I remember this has a funny history; see e.g. `.tx/config`.  I think Django uses the underscore format for its locale files, and so that's how we have things on the filesystem, or something?

Reply:
Actually this fixes them too, though havn't added backend condition for `id-id`. So updating again!

Reply:
@timabbott, You are right. i18next sends zh-Hans as zh-Hans while Django recognizes it as zh_Hans, see https://github.com/zulip/zulip/issues/6885 for more information.